jgrapht-core/src/main/java/org/jgrapht/graph/specifics/Specifics.java

```diff
@@ -76,10 +77,43 @@
 
     /**
      * Adds the specified edge to the edge containers of its source and target vertices.
-     *
+     * 
      * @param e the edge
+     * @deprecated Use method {@link #addEdgeToTouchingVertices(Object, Object, Object)} instead.
      */
+    @Deprecated
     void addEdgeToTouchingVertices(E e);
+    
+    /**
+     * Adds the specified edge to the edge containers of its source and target vertices.
+     * 
+     * @param sourceVertex the source vertex
+     * @param targetVertex the target vertex
+     * @param e the edge
+     * @return true if the edge was added, false otherwise
+     */
+    boolean addEdgeToTouchingVertices(V sourceVertex, V targetVertex, E e);
+
+    /**
+     * Adds a new edge to the edge containers of its source and target vertices.
+     * 
+     * @param sourceVertex the source vertex
+     * @param targetVertex the target vertex
+     * @param e the edge
+     * @return true if the edge was added, false otherwise
+     */
+    boolean addEdgeToTouchingVerticesIfAbsent(V sourceVertex, V targetVertex, E e);
+    
+    /**
+     * Adds a new edge to the edge containers of its source and target vertices.
+     * 
+     * @param sourceVertex the source vertex
+     * @param targetVertex the target vertex
+     * @param edgeSupplier the function which will create the edge
+     * @return the newly created edge or null if an edge with the same source and target vertices
+     *         was already present
+     */
+    E computeEdgeToTouchingVerticesIfAbsent(V sourceVertex, V targetVertex, Supplier<E> edgeSupplier);
```

Comment:
I think "create" instead of "compute" would be more natural.

Reply:
The `Map` interface contains a `computeIfAbsent`. I used the same name for consistency. I can change it if is sounds better.

Reply:
Let's change it.  (The "compute" in map indicates a more general operation.)